shenyu-loadbalancer/src/main/java/org/apache/shenyu/loadbalancer/spi/AbstractLoadBalancer.java

```diff
@@ -40,10 +41,15 @@ public Upstream select(final List<Upstream> upstreamList, final String ip) {
         if (CollectionUtils.isEmpty(upstreamList)) {
             return null;
         }
-        if (upstreamList.size() == 1) {
-            return upstreamList.get(0);
+        // available upstream list
+        List<Upstream> available = upstreamList.stream().filter(upstream -> getWeight(upstream) > 0).collect(Collectors.toList());
```

Comment:
If there are three upstreams in `DividePlugin`, and their weight is 0, maybe it's error here, right?

Reply:
Yes

Reply:
@ywengineer  this pr need update？

Reply:
no, return null if there isn't any available upstream, see line 47.

Reply:
if used HashLoadBalancer？ it not need weight~

Reply:
so, do you mean that we should use the status of upstream to determine it is available or not is better than weight?